crates/router/src/core/payment_link.rs

```diff
@@ -71,16 +70,27 @@ pub async fn intiate_payment_link_flow(
         .await
         .to_not_found_response(errors::ApiErrorResponse::PaymentLinkNotFound)?;
 
-    let payment_link_config = merchant_account
-        .payment_link_config
-        .map(|pl_config| {
+    let payment_link_config = if let Some(pl_config) = payment_link.payment_link_config.clone() {
```

Comment:
Instead of replicating the entire logic, we can get payment_link_config either from payment_link or merchant_account, and use common code for the rest